cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1580,6 +1834,80 @@ data TxBodyContent build era =
      }
      deriving (Eq, Show)
 
+instance
+  ( Default (BuildTxWith build (Maybe ProtocolParameters))
+  , IsCardanoEra era
+  ) => Default (TxBodyContent build era) where
+  def = TxBodyContent
+    { txIns = def
+    , txInsCollateral = def
+    , txInsReference = def
+    , txOuts = def
+    , txTotalCollateral = def
+    , txReturnCollateral = def
+    , txFee = def
+    , txValidityRange = (def, def)
+    , txMetadata = def
+    , txAuxScripts = def
+    , txExtraKeyWits = def
+    , txProtocolParams = def
+    , txWithdrawals = def
+    , txCertificates = def
+    , txUpdateProposal = def
+    , txMintValue = def
+    , txScriptValidity = def
+    }
+
+setTxIns :: TxIns build era -> TxBodyContent build era -> TxBodyContent build era
```

Comment:
I think you might want to restrict these "Builder functions" to the `BuildTx` case.

Also, we found `addXXX` be more useful on the lists/sets of things. It may very well be that you need to build the transaction step-by-step. Maybe that is something we can provide as a follow-up PR :)

Reply:
> I think you might want to restrict these "Builder functions" to the BuildTx case.

:+1: 
> Also, we found addXXX be more useful on the lists/sets of things.

We should also add these functions as there are probably use cases for both.